src/Swarm/TUI/Controller.hs

```diff
@@ -120,6 +124,14 @@ handleEvent s ev = do
 setFocus :: AppState -> Name -> EventM Name (Next AppState)
 setFocus s name = continue $ s & uiState . uiFocusRing %~ focusSetCurrent name
 
+toggleModal :: AppState -> Modal -> EventM Name (Next AppState)
+toggleModal s modal =
+  continue $ s
+      & gameState . paused %~ not
```

Comment:
First off, I totally agree the game should automatically pause if you open the help.  However, I'm not sure this works correctly. What if you pause the game and then hit F1?  Wouldn't that *unpause* the game and show the help modal?  We could fix that by setting the `paused` state based on the state of the modal.  But then there is another problem; what if you have manually paused the game and then open the help?  I would argue that the game should remain paused when you close the help.  So maybe `paused` can't be a `Bool` any more, but should be something like `data RunStatus = Running | ManualPause | AutoPause` so we can tell the difference.

Reply:
Good point, that was an oversight. We also needs to need to skip the event panel dispatch when a modal is displayed, e.g. to avoid accidental unpause while looking the help.

The last commit implements your RunStatus suggestion.